pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -147,26 +147,26 @@ buildHostCmd() {
     if [ -z "$buildHost" ]; then
         "$@"
     elif [ -n "$remoteNix" ]; then
-        ssh $SSHOPTS "$buildHost" "${maybeSudo[@]}" env PATH="$remoteNix":'$PATH' "$@"
+        ssh "${SSHOPTS[@]}" "$buildHost" "${maybeSudo[@]}" env PATH="$remoteNix":'$PATH' "$@"
```

Comment:
There’s another quoting bug in these `ssh` invocations: since `ssh` [smashes the remote command arguments together with spaces](https://www.chiark.greenend.org.uk/~cjwatson/blog/ssh-quoting.html), passing them as arrays doesn’t really help. They need an explicit escaping pass (and, for clarity, should then be passed to `ssh` as a single string):

```suggestion
        ssh "${SSHOPTS[@]}" "$buildHost" "${maybeSudo[*]@Q} env PATH=${remoteNix@Q}:\"\$PATH\" ${*@Q}"
```

Reply:
TIL two things:

- `@Q` basically does `printf '%q '` escaping.
- It does the "right thing" even when using `*` rather than `@` to index the whole array:

   ```
   $ bash --noprofile --norc -o xtrace
   bash-5.1$ maybeSudo=(printf '%q\n' 'open     spaces')
   + maybeSudo=(printf '%q\n' 'open     spaces')
   bash-5.1$ set -x -- 'more     spaceeee' 'wiiiide   ooopen    spaaaaces'
   + set -x -- 'more     spaceeee' 'wiiiide   ooopen    spaaaaces'
   bash-5.1$ diff <(ssh 127.0.0.1 "${maybeSudo[*]@Q} env PATH=\"\$PATH\" ${*@Q}") <(ssh 127.0.0.1 "${maybeSudo[@]@Q} env PATH=\"\$PATH\" ${@@Q}")
   + diff /dev/fd/63 /dev/fd/62
   ++ ssh 127.0.0.1 ''\''printf'\'' '\''%q\n'\'' '\''open     spaces'\'' env PATH="$PATH" '\''more     spaceeee'\'' '\''wiiiide   ooopen    spaaaaces'\'''
   ++ ssh 127.0.0.1 ''\''printf'\''' ''\''%q\n'\''' ''\''open     spaces'\'' env PATH="$PATH" '\''more     spaceeee'\''' ''\''wiiiide   ooopen    spaaaaces'\'''
   bash-5.1$ echo $?
   + echo 0
   0
   ```

Basically, the command is different but the result is the same whether you use `${*@Q}` or `${@@Q}`! Thank you!